server/src/main/java/org/opensearch/action/search/DeletePitRequest.java

```diff
@@ -0,0 +1,136 @@
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request to delete one or more PIT contexts based on IDs.
+ */
+public class DeletePitRequest extends ActionRequest implements ToXContentObject {
+
+    /**
+     * List of PIT IDs to be deleted , and use "_all" to delete all PIT reader contexts
+     */
+    private List<String> pitIds;
+
+    public DeletePitRequest(StreamInput in) throws IOException {
+        super(in);
+        pitIds = Arrays.asList(in.readStringArray());
+    }
+
+    public DeletePitRequest(String... pitIds) {
+        if (pitIds != null) {
+            this.pitIds = Arrays.asList(pitIds);
+        }
+    }
+
+    public DeletePitRequest(List<String> pitIds) {
+        if (pitIds != null) {
```

Comment:
Is null really valid input here? I'd consider just letting this fail with an NPE (or throwing an IllegalArgumentException).